### ISSUE TITLE
Regarding `command` backend Windows compatibility

### ISSUE THREAD

**Issue by quienn** [luarocks/luarocks#1450]
I've been struggling with the `command` backend, 'cause it seems like it's not working on Windows.

From most searches I've done regarding this functionality, this should technically work, but it doesn't:

![image](https://user-images.githubusercontent.com/86124825/196582721-56eda534-f25d-4d34-bb1a-60473a10889c.png)

I've tried both ways (wrapping it around the platform overrides), and both won't do anything. It just prints the `build_command`.

![image](https://user-images.githubusercontent.com/86124825/196583033-d24ede7f-8529-4901-8a9b-e3482bbc5a25.png)

Hope someone can help me figure out why is it not working, or if this is intended (a quick glimpse at the backend source suggests that it should *just work*), when will it be available for Windows developers?

Edit: I installed Lua through hererocks, using the `mingw` target.
```text
$ luarocks --version
C:\Users\quien\Dev\Lua\5.4\luarocks\luarocks.lua 3.9.1
```

**Comment by alerque** [comment]
I doubt it is just *printing* the command, I assume it is echoing it showing you what it is executing. As such I'd guess it is getting run already and just not accomplishing what you expect. Try some other command like a print and see if the output gets shown. Also try checking if the CWD is what you expect it is at run time.

**Comment by hishamhm** [comment]
The "luarocks build" command runs in a separate sandbox folder, not in the
current folder. "luarocks make" runs on the current folder, that command is
meant for development time.

**Comment by quienn** [comment]
It's exactly the same behaviour with `luarocks make`, though. How can I debug and check if it's actually doing something if there's no output? 🤔

**Comment by quienn** [comment]
![image](https://user-images.githubusercontent.com/86124825/196860168-4773bf10-4734-47e9-94e6-f71d76b8f729.png)


It will always "succeed", even if the command does not exist.

**Comment by Tieske** [comment]
to debug you can try and use the `--verbose` flag to see what commands it invokes

**Comment by quienn** [comment]
This is the output I get with `.\luarocks.bat make --verbose`:
```lua
cal %CC% -- This is where the value of `build_command` is printed when doing regular `luarocks make`
fs.export_cmd("CC", "C:\\Users\\quien\\scoop\\apps\\msys2\\current\\mingw64\\bin\\gcc.exe")
fs.execute_string("SET CC=C:\\Users\\quien\\scoop\\apps\\msys2\\current\\mingw64\\bin\\gcc.exe\
cal %CC%")
fs.current_dir()
fs.command_at("C:\\Users\\quien\\Dev\\Proyectos\\ef", "SET CC=C:\\Users\\quien\\scoop\\apps\\msys2\\current\\mingw64\\bin\\gcc.exe\
cal %CC%")
fs.Q("C:\\Users\\quien\\Dev\\Proyectos\\ef")

os.execute:     C: & cd "C:\Users\quien\Dev\Proyectos\ef" & SET CC=C:\Users\quien\scoop\apps\msys2\current\mingw64\bin\gcc.exe
cal %CC%
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0
```

Even if the command is invalid or it just fails, it doesn't complain about it. Here's another example:
```
>  .\luarocks make
ef dev-1 depends on tl ~> 0.14.1 (0.14.1-1 installed)

ef dev-1 depends on penlight ~> 1.13.1 (1.13.1-1 installed)
exit /b 1
ef dev-1 is now installed in C:\Users\quien\Dev\Proyectos\ef/lua_modules (license: MIT)
```

**Comment by hishamhm** [comment]
From the output above, perhaps the problem is that command on Windows in being concatenated to its preliminary `C: & SET=...` command with a newline, rather than a ` & `?

**Comment by hishamhm** [comment]
Fixed by #1569!
